lib/ansible/plugins/strategy/__init__.py

```diff
@@ -385,6 +391,32 @@ def get_delegated_hosts(self, result, task):
         host_name = result.get('_ansible_delegated_vars', {}).get('ansible_delegated_host', None)
         return [host_name or task.delegate_to]
 
+    def _set_always_delegated_facts(self, result, task):
+        """Sets host facts for ``delegate_to`` hosts for facts that should
+        always be delegated
+
+        See ``ALWAYS_DELEGATE_FACTS``
+        """
+        if task.delegate_to is None:
+            return
+
+        facts = result['ansible_facts']
+        always_keys = set()
```

Comment:
This one screams (to me at least) for a nested generator- I was having a hard time following the aliased function stuff.

```suggestion:
        delegated_keys = set((k for k in facts if any((k.startswith(prefix) for prefix in ALWAYS_DELEGATE_FACTS))))
```

Reply:
I actually did that the first go around. And then I decided to test it for performance, and surprisingly the nested generator expression turned out to be slower.

Reply:
The aliased function stuff would be just as useful in a nested generator if possible.  I don't know where the post is that explains this, but this has started becoming more common in the cpython code base. It eliminates extra attribute lookups, which improves performance.